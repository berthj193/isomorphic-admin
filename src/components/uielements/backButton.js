import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

import Button from '../uielements/button';

export class GoBackButton extends PureComponent {
  static propTypes = {
    history: PropTypes.shape({
      goBack: PropTypes.func.isRequired,
    }).isRequired,
  };

  goBack = () => this.props.history.goBack();

  render() {
    return (
      <Button onClick={this.goBack}>
          Go Back
      </Button>
    );
  }
}

export default withRouter(GoBackButton);
