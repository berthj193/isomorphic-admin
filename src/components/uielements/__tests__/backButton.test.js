import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GoBackButton } from '../backButton';

jest.mock('react-router-dom', () => ({
  // eslint-disable-next-line react/display-name
  Link: () => <a />,
  withRouter: () => null,
}));

describe('<BackButton />', () => {
  const props = {
    history: {
      goBack: jest.fn(),
    },
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('should render properly', () => {
    expect(toJson(shallow(<GoBackButton {...props} />))).toMatchSnapshot();
  });

  it('should run goBack method on click', () => {
    mount(<GoBackButton {...props} />)
      .find('button')
      .first()
      .simulate('click');
    expect(props.history.goBack).toHaveBeenCalledTimes(1);
  });
});
