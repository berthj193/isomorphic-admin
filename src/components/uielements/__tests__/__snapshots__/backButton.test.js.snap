// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackButton /> should render properly 1`] = `
<Button
  ghost={false}
  loading={false}
  onClick={[Function]}
  prefixCls="ant-btn"
>
  Go Back
</Button>
`;
